persistent/Database/Persist/Sql/Run.hs

```diff
@@ -28,15 +86,16 @@ import Database.Persist.Sql.Raw
 runSqlPool
     :: (MonadUnliftIO m, BackendCompatible SqlBackend backend)
     => ReaderT backend m a -> Pool backend -> m a
-runSqlPool r pconn = withRunInIO $ \run -> withResource pconn $ run . runSqlConn r
+runSqlPool r pconn = with (acquireSqlConnFromPool pconn) $ runReaderT r
 
 -- | Like 'runSqlPool', but supports specifying an isolation level.
 --
 -- @since 2.9.0
 runSqlPoolWithIsolation
     :: (MonadUnliftIO m, BackendCompatible SqlBackend backend)
     => ReaderT backend m a -> Pool backend -> IsolationLevel -> m a
-runSqlPoolWithIsolation r pconn i = withRunInIO $ \run -> withResource pconn $ run . (\conn -> runSqlConnWithIsolation r conn i)
+runSqlPoolWithIsolation r pconn i =
+    with (acquireSqlConnFromPoolWithIsolation i pconn) $ runReaderT r
```

Comment:
Same here - `runSqlConn` does all the transaction stuff, so `runReaderT` here voids that safety. (Man I really need to `newtype` SqlPersistT ...)

```suggestion
    with (acquireSqlConnFromPoolWithIsolation i pconn) $ \conn -> runSqlConnWithIsolation r conn i
```

Reply:
Wait, no, I think I see what's going on here. You've moved that responsibility into the `Acquire` interface. So `runSqlConn` has it's own logic for handling exceptions, and that logic is now sort of duplicated into the `Acquire` interface.

This is fine. Ignore the suggestions. I've reviewed the relevant code and as far as I can tell the behavior will be identical.